django/contrib/admin/templatetags/admin_list.py

```diff
@@ -167,6 +167,7 @@ def make_qs_param(t, n):
 
         yield {
             "text": text,
+            "id": slugify(text),
```

Comment:
I can see a few changes needed here:

- We should derive the label from the `field_name` rather than slugified text. This will be more consistent with other aspects of the forms framework
- There should be a prefix to the id, to reduce the risk of conflicts with other parts of the page.

First draft:

And possibly account for multiple changelist tables in the same page? I don’t know if this is possible in Django’s admin?

Reply:
> And possibly account for multiple changelist tables in the same page? I don’t know if this is possible in Django’s admin?

I don't think this is possible 

I would also suggest as an alternative to use the index for the id (so `cl-header-0`, `cl-header-1` etc) as the index is also available here and also when rendering the data (`field_index`)